packages/calcite-components/src/components/navigation-user/navigation-user.scss

```diff
@@ -24,28 +23,29 @@
       transition-default
       focus-base
       font-sans;
-    background-color: var(--calcite-navigation-user-background-color, transparent);
+    background-color: var(
+      --calcite-navigation-user-background-color,
+      var(--calcite-internal-navigation-user-background-color, transparent)
+    );
     border: none;
-    border-block-end-width: var(--calcite-spacing-base);
-    border-block-end-style: solid;
-    border-color: var(--calcite-navigation-user-border-color, transparent);
+    border-block-end: var(--calcite-spacing-base) solid
+      var(--calcite-navigation-user-border-color, var(--calcite-internal-navigation-user-border-color, transparent));
     font-size: var(--calcite-font-size-md);
     line-height: var(--calcite-font-line-height-fixed-lg);
   }
 }
 
 :host(:hover) .button,
 :host(:focus) .button {
-  --calcite-navigation-user-background-color: var(--calcite-color-foreground-2);
+  --calcite-internal-navigation-user-background-color: var(--calcite-color-foreground-2);
 }
 
 :host(:focus) .button {
   @apply focus-inset;
 }
 
 :host(:active) .button {
-  --calcite-navigation-user-background-color: var(--calcite-color-foreground-3);
-  --calcite-navigation-user-text-color: var(--calcite-color-text-1);
```

Comment:
I might be wrong but I think you want to add this here?
`--calcite-internal-navigation-user-text-color: var(--calcite-color-text-1);`